src/pal/src/loader/module.cpp

```diff
@@ -92,14 +93,15 @@ CRITICAL_SECTION module_critsec;
 MODSTRUCT exe_module; 
 MODSTRUCT *pal_module = NULL;
 
-char g_szCoreCLRPath[MAX_LONGPATH] = { 0 };
+char * g_szCoreCLRPath = new char[MAX_LONGPATH+1];
```

Comment:
This change is not necessary and doesn't bring any benefit. What would help though is to assing NULL to the variable here and change the code that assigns to it to perform the allocation (and add assert it is not NULL in PAL_BindResources).